objectRecognitionSource/src/main/java/io/vantiq/extsrc/objectRecognition/ObjectRecognitionCore.java

```diff
@@ -79,7 +80,8 @@
     private static final String UPLOAD = "upload";
     private static final String DELETE = "delete";
     private static final String FILTER = "filter";
-    
+    final static String SYNCH_KEY = "synchKey";
+
```

Comment:
since this is a static, this is single threading for ALL instances of ObjectRecognitionCore.  That's probably OK, but that's what it's doing.

Since it seems to be used to sync to get local copy of the neuralNet variable (which is an instance variable), making this static made the synchronization more draconian.

In truth, if I've read everything right, sync(this) is fine here.  Or make SYNC_KEY non-static, and change the name to NEURAL_NET_SYNC or something.  & Add a comment re: what it's doing.

Reply:
Understood. I think the simplest option would be to just remove the SYNCH_KEY and  synchronize on (this) instead.